src/inmanta/agent/executor.py

```diff
@@ -164,9 +173,9 @@ def blueprint_hash(self) -> str:
         if self._hash_cache is None:
             blueprint_dict = {
                 "pip_config": self.pip_config.dict(),
-                "requirements": sorted(self.requirements),
+                "requirements": self.requirements,
                 # Use the hash values of the sources, sorted to ensure consistent ordering
-                "sources": sorted(source.hash_value for source in self.sources),
+                "sources": sorted(list(set(source.hash_value for source in self.sources))),
```

Comment:
I'm not convinced we want to remove duplicates here. We might end up with different executors that have the same hash, e.g. if one of them contains an additional empty `__init__.py` file somewhere.